File: TsinghuaNet/FluxUser.cs
using System;

namespace TsinghuaNet
{
    /// <summary>
    /// A simple structure represents the current user online.
    /// </summary>
    public readonly struct FluxUser
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="FluxUser"/> class.
        /// </summary>
        /// <param name="username">Username of the user.</param>
        /// <param name="flux">Flux used by the user this month.</param>
        /// <param name="onlineTime">Online time used this time of the user.</param>
        /// <param name="balance">The network balance of the user.</param>
        public FluxUser(string username, long flux, TimeSpan onlineTime, decimal balance)
        {
            Username = username;
            Flux = flux;
            OnlineTime = onlineTime;
            Balance = balance;
        }
        /// <summary>
        /// Username of the user.
        /// </summary>
        public string Username { get; }
        /// <summary>
        /// Flux used by the user this month.
        /// </summary>
        public long Flux { get; }
        /// <summary>
        /// Online time used this time of the user.
        /// </summary>
        public TimeSpan OnlineTime { get; }
        /// <summary>
        /// The network balance of the user.
        /// </summary>
        public decimal Balance { get; }

        public static bool operator ==(FluxUser u1, FluxUser u2) => u1.Username == u2.Username;
        public static bool operator !=(FluxUser u1, FluxUser u2) => !(u1 == u2);

        /// <summary>
        /// Determines whether the username of the two <see cref="FluxUser"/> are equal.
        /// </summary>
        /// <param name="obj">The other object.</param>
        /// <returns><see langword="true"/> if they're equal; otherwise, <see langword="false"/>.</returns>
        public override bool Equals(object obj)
        {
            if (obj is FluxUser other)
            {
                return this == other;
            }
            return false;
        }
        /// <summary>
        /// Returns the hash value of this object.
        /// </summary>
        /// <returns>The hash value.</returns>
        public override int GetHashCode()
        {
            return Username?.GetHashCode() ?? 0;
        }

        internal static FluxUser Parse(string fluxstr)
        {
            string[] r = fluxstr.Split(',');
            if (string.IsNullOrWhiteSpace(r[0]))
            {
                return default;
            }
            else
            {
                return new FluxUser(
                    r[0],
                    long.Parse(r[6]),
                    TimeSpan.FromSeconds(long.Parse(r[2]) - long.Parse(r[1])),
                    decimal.Parse(r[11]));
            }
        }
    }
}

File: TsinghuaNet/FluxUser.vb
''' <summary>
''' 表示在线用户信息
''' </summary>
Public Structure FluxUser
    ''' <summary>
    ''' 用户名
    ''' </summary>
    Public ReadOnly Property Username As String
    ''' <summary>
    ''' 使用流量
    ''' </summary>
    Public ReadOnly Property Flux As ByteSize
    ''' <summary>
    ''' 在线时长
    ''' </summary>
    Public ReadOnly Property OnlineTime As TimeSpan
    ''' <summary>
    ''' 剩余流量
    ''' </summary>
    Public ReadOnly Property Balance As Decimal

    ''' <summary>
    ''' 使用相应信息初始化<see cref="FluxUser"/>
    ''' </summary>
    ''' <param name="username">用户名</param>
    ''' <param name="flux">使用流量</param>
    ''' <param name="onlineTime">在线时长</param>
    ''' <param name="balance">剩余流量</param>
    Public Sub New(username As String, flux As ByteSize, onlineTime As TimeSpan, balance As Decimal)
        Me.Username = username
        Me.Flux = flux
        Me.OnlineTime = onlineTime
        Me.Balance = balance
    End Sub

    Public Shared Operator =(u1 As FluxUser, u2 As FluxUser) As Boolean
        Return u1.Username = u2.Username AndAlso u1.Flux = u2.Flux AndAlso u1.OnlineTime = u2.OnlineTime AndAlso u1.Balance = u2.Balance
    End Operator

    Public Shared Operator <>(u1 As FluxUser, u2 As FluxUser) As Boolean
        Return Not u1 = u2
    End Operator

    ''' <inheritdoc/>
    Public Overrides Function Equals(obj As Object) As Boolean
        Return TypeOf obj Is FluxUser AndAlso Me = CType(obj, FluxUser)
    End Function

    ''' <inheritdoc/>
    Public Overrides Function GetHashCode() As Integer
        Return If(Username?.GetHashCode(), 0)
    End Function

    ''' <summary>
    ''' 转换字符串为<see cref="FluxUser"/>实例
    ''' </summary>
    ''' <param name="fluxstr">字符串</param>
    ''' <returns>实例</returns>
    Friend Shared Function Parse(fluxstr As String) As FluxUser
        Dim r = fluxstr.Split(","c)
        If String.IsNullOrWhiteSpace(r(0)) Then
            Return Nothing
        Else
            Return New FluxUser(
                r(0),
                ByteSize.FromBytes(r(6)),
                TimeSpan.FromSeconds(CLng(r(2)) - CLng(r(1))),
                r(11))
        End If
    End Function
End Structure
